search-sentiments-map/public/scripts/modal-script.js

```diff
@@ -91,18 +92,48 @@ function displayTopResultsForCurrentTrend(countryCode) {
         countryData[0].interest + '</b><br>';
     resultElement.innerHTML += '<b>Average Sentiment Score: ' + 
         countryData[0].averageSentiment.toFixed(1) + '</b><br>';
-
+    
     // Get search results for the specified country.
     let results = countryData[0].results;
     for (let i = 0; i < results.length; i++) {
-      // i + 1 shows the index for each search result. 
-      resultElement.innerHTML += (i + 1).toString() + '. ' + '<a href=' + 
-          results[i].link + '>' + results[i].htmlTitle + '</a><br>' + 
-          results[i].snippet + '<br>' + '<i>Sentiment Score: ' + 
-          results[i].score.toFixed(1) + '</i><br>';
+      await fetch('/sentiment-words/' + results[i].title + results[i].snippet)
+          .then(resultsJsonArray => resultsJsonArray.json())
+          .then(sentimentWordsResult => {
+            // i + 1 shows the index for each search result.
+            resultElement.innerHTML += (i + 1).toString() + '. ' + '<a href=' + 
+                results[i].link + '>' + results[i].htmlTitle + '</a><br>';
+            // Check to make sure that there is detection of positive and 
+            // negative words from the Node.js sentiment API.
+            if (sentimentWordsResult != null && 
+                sentimentWordsResult.positive.length != 0) {
+              var snippetArray = results[i].snippet.split(" ");
+              snippetArray.forEach((word) => {
+                // Replace quotations or other symbols that may be attached to 
+                // the word.  
+                let checkWordInArray = word.replace(/[^a-z0-9+]+/gi, '');
+                if (results[i].score > 0 && sentimentWordsResult.positive
+                    .includes(checkWordInArray)) {
+                  resultElement.innerHTML += '<span style=\'color: ' + 
+                      POSITIVE_COLOR + ';\'>' + word + ' </span>';
+                } else if (results[i].score < 0 && sentimentWordsResult.negative
+                    .includes(checkWordInArray)) {
```

Comment:
Have you considered looping through the positive or negative words rather than every word in the snippet? Because then you only loop through the words you know need to change.

Reply:
Yea I thought about that possibility before coding it, but I don't think there's then a way to singly color a word in the snippet without going through each word, bc each word could be a different color

Reply:
Is there any javascript function for find and replace elements in an array so that you can find the words that you look for and replace them with their colored version

Reply:
Or could you split the string at that word and then concatenate the strings back together with the new colored word?

Reply:
yeah that's also what I mean

Reply:
+1 to the easier approaches using existing util functions.